lib/color-scales.js

```diff
@@ -0,0 +1,85 @@
+export class InterpolatingColorScale {
+
+  // Pass in an array of [[z, r, g, b, a], [z, r, g, b, a], [z, r, g, b, a]...]
+  constructor (breaks) {
+    this.breaks = breaks
+    this.interpolate = true
+  }
+
+  getColorForValue (value) {
+    var currBreak, prevBreak
+    prevBreak = this.breaks[0]
+    for (currBreak of this.breaks) {
+      if (value < currBreak[0]) {
+        if (this.interpolate) {
+          const [z0, r0, g0, b0, a0] = prevBreak
+          const [z1, r1, g1, b1, a1] = currBreak
+          const frac = (value - z0) / (z1 - z0)
+          const r = r0 + (r1 - r0) * frac
+          const g = g0 + (g1 - g0) * frac
+          const b = b0 + (b1 - b0) * frac
+          const a = a0 + (a1 - a0) * frac
+          return [r, g, b, a]
+        } else {
+          const [z, r, g, b, a] = currBreak
+          return [r, g, b, a]
+        }        
+      }
+      prevBreak = currBreak
+    }
+    const [z, r, g, b, a] = prevBreak
+    return [r, g, b, a]
+  }
+  
+}
+
+export class CutoffColorScale {
```

Comment:
Using classes for everything is definitely a Java convention :)

The JS-way<sup>TM</sup>:

```js
const cutoffColorScale = (cutoff, lowColor, highColor) =>
  (value) => value < cutoff ? lowColor : highColor
```

This goes for almost all of the classes here assuming you don't need to change the cutoffs or colors on the fly. If you do, might as well just execute again to get the new color function.